File: 백준/Bronze/29736. 브실이와 친구가 되고 싶어 🤸‍♀️/브실이와 친구가 되고 싶어 🤸‍♀️.cs
string[] str = Console.ReadLine().Split();

int A = int.Parse(str[0]);
int B = int.Parse(str[1]);

str = Console.ReadLine().Split();
int K = int.Parse(str[0]);
int X = int.Parse(str[1]);

int people = 0;
for (int i = K-X; i < K+X+1; i++)
{
    if (i < A || i > B) continue;
    people++;
}

if (people == 0) 
    Console.WriteLine("IMPOSSIBLE");
else
    Console.WriteLine(people);

File: 백준/Bronze/29736. 브실이와 친구가 되고 싶어 🤸‍♀️/브실이와 친구가 되고 싶어 🤸‍♀️.py

A, B = map(int, input().split())


K, X = map(int, input().split())

people = 0
for i in range(K - X, K + X + 1):
    if i < A or i > B:
        continue
    people += 1

if people == 0:
    print("IMPOSSIBLE")
else:
    print(people)
